op-supervisor/supervisor/backend/syncnode/node.go

```diff
@@ -326,8 +319,16 @@ func (m *ManagedNode) resetSignal(errSignal error, l1Ref eth.BlockRef) {
 		if err != nil {
 			m.log.Warn("Node failed to reset", "err", err)
 		}
-	case types.ErrOutOfOrder:
+	case errors.Is(errSignal, types.ErrOutOfOrder):
+		s, err := m.backend.LocalSafe(ctx, m.chainID)
+		if err != nil {
+			m.log.Warn("Failed to retrieve local-safe", "err", err)
+		}
```

Comment:
If it errors it should return here, the returned data is not safe to continue to use, likely just zeroed.

Reply:
In fact, i think the function needs early returns across the board, I agree.